core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
The x86 code has an early return from `decode_opcode`, before the call to `instr_set_raw_bits`, but the x86 code does not call `instr_get_opcode` from `instr_encode_arch` before testing `instr_raw_bits_valid`. An awful lot of twiddling of `INSTR_RAW_BITS_VALID` can occur because of an innocent-looking `instr_get_opcode`...